rdk/rdk.py

```diff
@@ -1337,6 +1552,340 @@ def deploy(self):
+                    'ParameterKey': 'BoundaryPolicyArn',
+                    'ParameterValue': boundaryPolicyArn,
+                },
+                {
+                    'ParameterKey': 'SourceBucket',
+                    'ParameterValue': code_bucket_name,
+                },
+                {
+                    'ParameterKey': 'SourcePath',
+                    'ParameterValue': s3_dst,
+                },
+                {
+                    'ParameterKey': 'SourceRuntime',
+                    'ParameterValue': self.__get_runtime_string(rule_params),
+                },
+                {
+                    'ParameterKey': 'SourceEvents',
+                    'ParameterValue': source_events,
+                },
+                {
+                    'ParameterKey': 'SourcePeriodic',
+                    'ParameterValue': source_periodic,
+                },
+                {
+                    'ParameterKey': 'SourceInputParameters',
+                    'ParameterValue': json.dumps(combined_input_parameters),
+                },
+                {
+                    'ParameterKey': 'SourceHandler',
+                    'ParameterValue': self.__get_handler(rule_name, rule_params)
+
+                },
+                {
+                    'ParameterKey': 'Timeout',
+                    'ParameterValue': str(self.args.lambda_timeout)
+                }]
+            layers = []
+            if 'SourceRuntime' in rule_params:
+                if rule_params['SourceRuntime'] in ['python3.6-lib', 'python3.7-lib', 'python3.8-lib', 'python3.9-lib']:
+                    if self.args.rdklib_layer_arn:
+                        layers.append(self.args.rdklib_layer_arn)
+                    else:
+                        rdk_lib_version = RDKLIB_LAYER_VERSION[my_session.region_name]
+                        rdklib_arn = RDKLIB_ARN_STRING.format(region=my_session.region_name, version=rdk_lib_version)
+                        layers.append(rdklib_arn)
+
+
+            if self.args.lambda_layers:
+                additional_layers = self.args.lambda_layers.split(',')
+                layers.extend(additional_layers)
+
+            if layers:
+                my_params.append({
+                    'ParameterKey': 'Layers',
+                    'ParameterValue': ",".join(layers)
+                })
+
+            if self.args.lambda_security_groups and self.args.lambda_subnets:
+                my_params.append({
+                    'ParameterKey': 'SecurityGroupIds',
+                    'ParameterValue': self.args.lambda_security_groups
+                })
+                my_params.append({
+                    'ParameterKey': 'SubnetIds',
+                    'ParameterValue': self.args.lambda_subnets
+                })
+
+            #create json of CFN template
+            cfn_body = os.path.join(path.dirname(__file__), 'template',  "configRuleOrganization.json")
+            template_body = open(cfn_body, "r").read()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *open*. The resource is allocated by call *builtins.open*. Currently, there are execution paths that do not contain closure statements, for example, when *TextIOWrapper.read()* throws an exception. To prevent this resource leak, close the object returned by *open()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).